api/src/test/java/org/openmrs/OpenmrsTestsTest.java

```diff
@@ -209,4 +218,49 @@ public void shouldHaveClassNameEndWithTestIfContainsMethodTestAnnotations() {
 		return currentDirTestClasses;
 	}
 	
+	/**
+	 * Make sure all tests in this class have a call to assert in them. This would help prevent
+	 * people from making tests that just print results to the screen
+	 * 
+	 * @throws CannotCompileException
+	 * @throws Throwable
+	 * @throws Exception
+	 */
+	@Test
+	public void shouldHaveAtLeastOneCallToAssert() throws Throwable {
+		ClassPool cp = ClassPool.getDefault();
+		CtClass ctClass = cp.get("org.openmrs.OpenmrsTestsTest");
+		
+		//get all methods called in "org.openmrs.OpenmrsTestsTest" class
+		CtMethod[] cMethod = ctClass.getDeclaredMethods();
```

Comment:
It seems to me that this is only looking at the methods in OpenmrsTestsTest. Is the goal not to prevent tests in the openmrs api package to each have an assertion?

Reply:
Yea, sure! the goal of this issue is to have all methods in openmrs-api-test call any assert.  But certain packages  in Openmrs-api eg org.openmrs.annotation  don't have a call assert.  I also thought about this as i was writing code.  I sent this PR to to seek thoughts on how best to skip certain packages.

Reply:
I  have some doubts about the usefulness of this task in general and added them in the JIRA issue. I would like to get some input from other community members if we should even implement this. Let's discuss this in the JIRA issue.